webui/master/src/containers/pages/Configuration/Configuration.tsx

```diff
@@ -82,6 +135,30 @@ export class ConfigurationPresenter extends React.Component<AllProps> {
       </div>
     );
   }
+
+  private __searchInputHandler = (e: React.ChangeEvent<HTMLInputElement>): void => {
```

Comment:
since the filtering isn't reliant on making any calls to the backend, i think it may worthwhile to do the filtering in realtime as the user types. That'll help simplify the logic, and you wouldn't need to keep track of the clearSearch button status since it'd just be dependent on if there's any input in the search bar atm. ideally you'd still trim and make the input lower-case only (could be done with regex or converted in the inputHandler).

what do you think about doing it this way?

Reply:
Yeah, I think this is a better approach. I think better user experience too.